bokeh/query.py

```diff
@@ -41,7 +41,7 @@ class LEQ(object): pass
 class NEQ(object): pass
 
 
-def match(obj, selector, context={}):
+def match(obj, selector, context=None):
```

Comment:
On the other hand, it seems more natural if context is used during recursion. I'll add it. Let me know if you think it should not.